src/main/java/com/mongodb/hibernate/internal/type/ObjectIdJdbcType.java

```diff
@@ -0,0 +1,125 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.type;
+
+import com.mongodb.hibernate.internal.FeatureNotSupportedException;
+import java.io.Serial;
+import java.sql.CallableStatement;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import org.bson.types.ObjectId;
+import org.hibernate.type.descriptor.ValueBinder;
+import org.hibernate.type.descriptor.ValueExtractor;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaType;
+import org.hibernate.type.descriptor.jdbc.BasicBinder;
+import org.hibernate.type.descriptor.jdbc.BasicExtractor;
+import org.hibernate.type.descriptor.jdbc.JdbcType;
+
+public final class ObjectIdJdbcType implements JdbcType {
+    @Serial
+    private static final long serialVersionUID = 1L;
+
+    private static final MqlType MQL_TYPE = MqlType.OBJECT_ID;
+    private static final ObjectIdJavaType JAVA_TYPE = ObjectIdJavaType.INSTANCE;
+    public static final ObjectIdJdbcType INSTANCE = new ObjectIdJdbcType();
+
+    private ObjectIdJdbcType() {}
+
+    @Override
+    public int getJdbcTypeCode() {
+        return MQL_TYPE.getVendorTypeNumber();
+    }
+
+    @Override
+    public String getFriendlyName() {
+        return MQL_TYPE.getName();
+    }
+
+    @Override
+    public <X> ValueBinder<X> getBinder(JavaType<X> javaType) {
+        if (!javaType.equals(JAVA_TYPE)) {
+            throw new FeatureNotSupportedException();
+        }
+        @SuppressWarnings("unchecked")
+        var result = (ValueBinder<X>) Binder.INSTANCE;
+        return result;
+    }
+
+    @Override
+    public <X> ValueExtractor<X> getExtractor(JavaType<X> javaType) {
+        if (!javaType.equals(JAVA_TYPE)) {
+            throw new FeatureNotSupportedException();
+        }
+        @SuppressWarnings("unchecked")
+        var result = (ValueExtractor<X>) Extractor.INSTANCE;
+        return result;
+    }
+
+    private static final class Binder extends BasicBinder<ObjectId> {
+        @Serial
+        private static final long serialVersionUID = 1L;
+
+        static Binder INSTANCE = new Binder();
+
+        private Binder() {
+            super(JAVA_TYPE, ObjectIdJdbcType.INSTANCE);
+        }
+
+        @Override
+        protected void doBind(PreparedStatement st, ObjectId value, int index, WrapperOptions options)
+                throws SQLException {
+            st.setObject(index, value, getJdbcType().getJdbcTypeCode());
+        }
+
+        @Override
+        protected void doBind(CallableStatement st, ObjectId value, String name, WrapperOptions options)
+                throws SQLException {
+            throw new SQLFeatureNotSupportedException();
+        }
```

Comment:
for completeness, we might need to implement
```
doBindNull();
```
methods to comply with JDBC API.

Reply:
nvm. saw the default implementation has covered that already.

Reply:
Since there is nothing to do here, could you resolve the discussion?

P.S. Note that this is a draft PR. You are welcome to review if you want, but a draft PR may either lack code that is needed, or have code that is not going to be present when it's ready for review.